src/commands/trace/trace.ts

```diff
@@ -0,0 +1,121 @@
+            See README for details.
+        `,
+    examples: [
+      ['Trace a command and report to Datadog', 'datadog-ci trace echo "Hello World"'],
+      [
+        'Trace a command and report to the datadoghq.eu site',
+        'DATADOG_SITE=datadoghq.eu datadog-ci trace echo "Hello World"',
+      ],
+    ],
+  })
+
+  private command?: string[]
+
+  private config = {
+    apiKey: process.env.DATADOG_API_KEY || process.env.DD_API_KEY,
+    envVarTags: process.env.DD_TAGS,
+  }
+
+  public async execute() {
+    if (!this.command || !this.command.length) {
+      this.context.stderr.write('Missing command to run\n')
+
+      return 1
+    }
+
+    const [command, ...args] = this.command
+    const id = crypto.randomBytes(5).toString('hex')
+    const t0 = performance.now()
+    const spawnResult = spawnSync(command, args, {env: {...process.env, DD_CUSTOM_PARENT_ID: id}, stdio: 'inherit'})
+    const t1 = performance.now()
+    const duration = t1 - t0
+    const exitCode = spawnResult.status ?? this.signalToNumber(spawnResult.signal!) ?? 127
+    const api = this.getApiHelper()
+    const [provider, data] = this.getData()
+    await api.reportCustomSpan(
+      {
+        data,
+        duration,
+        id,
+        is_error: exitCode !== 0,
+        parent_id: process.env.DD_CUSTOM_PARENT_ID,
+        tags: this.config.envVarTags ? parseTags(this.config.envVarTags.split(',')) : {},
+      },
+      provider
+    )
+
+    return exitCode
+  }
+
+  public getData(): [string, Record<string, string>] {
+    if (process.env.CIRCLECI === 'true') {
+      return [
+        'circleci',
+        this.getEnvironmentVars([
+          'CIRCLE_BRANCH',
+          'CIRCLE_BUILD_NUM',
+          'CIRCLE_BUILD_URL',
+          'CIRCLE_JOB',
+          'CIRCLE_NODE_INDEX',
+          'CIRCLE_NODE_TOTAL',
+          'CIRCLE_PROJECT_REPONAME',
+          'CIRCLE_PULL_REQUEST',
+          'CIRCLE_REPOSITORY_URL',
+          'CIRCLE_SHA1',
+          'CIRCLE_TAG',
+          'CIRCLE_WORKFLOW_ID',
+        ]),
+      ]
+    }
+    throw new Error('Cannot detect any CI Provider. This command only works if run as part of your CI.')
+  }
+
+  private getApiHelper(): APIHelper {
+    if (!this.config.apiKey) {
+      this.context.stdout.write(
+        `Neither ${chalk.red.bold('DATADOG_API_KEY')} nor ${chalk.red.bold('DD_API_KEY')} is in your environment.\n`
+      )
+      throw new Error('API key is missing')
+    }
+
+    return apiConstructor(this.getBaseIntakeUrl(), this.config.apiKey)
+  }
+
+  private getBaseIntakeUrl() {
+    if (process.env.DATADOG_SITE || process.env.DD_SITE) {
+      return `https://webhooks-http-intake.logs.${process.env.DATADOG_SITE || process.env.DD_SITE}`
+    }
+
+    return 'https://webhooks-http-intake.logs.datadoghq.com'
+  }
+
+  private getEnvironmentVars(keys: string[]): Record<string, string> {
+    return Object.fromEntries(keys.filter((key) => key in process.env).map((key) => [key, process.env[key]!]))
```

Comment:
`Object.fromEntries` is not available in `10.24.1`, which is the oldest version we support, so we can't use it. 

```suggestion
    return keys.reduce((envVarsDict, key) => {
       if (process.env[key]) {
         envVarsDict[key] = process.env[key]
       }
       return envVarsDict
    })
```